File: Binary-Tree-Inorder-Traversal/Binary-Tree-Inorder-Traversal.cpp

/**
 * Definition for binary tree
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    vector<int> inorderTraversal(TreeNode *root) {
        stack<TreeNode*>st;
        vector<int>ans;
        while(!st.empty() || root)
        {
            if(root)
            {
                st.push(root);
                root=root->left;
            }
            else
            {
                root=st.top();
                st.pop();
                ans.push_back(root->val);
                root=root->right;
            }
        }
        return ans;
    }
};


 

File: solutions/Binary-Tree-Inorder-Traversal/Binary-Tree-Inorder-Traversal.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number[]}
 */
var inorderTraversal = function (root) {
  const st = [];
  const ans = [];
  while (st.length !== 0 || root) {
    if (root) {
      st.push(root);
      root = root.left;
    }
    else {
      root = st.pop();
      ans.push(root.val);
      root = root.right;
    }
  }
  return ans;
};